docs/sources/community/lids/_DRAFT_AsynchronousQueries.md

```diff
@@ -0,0 +1,97 @@
+
+## Problem Statement
+
+Expensive/slow queries can be expressed in Loki terms in many different ways with various effects to the user experience, i.e.:
+
+* A query that has a too-wide label selector might require to scan a tremendous amount of chunks (i.e. effectively TB of logs or more) (e.g. parsing logs across a fleet of clusters).
+* A query that aggregates metrics over a very large time range (i.e. multiple months) and/or chunks (i.e. TB of logs or more) (e.g. count audit log events over months).
+* A query using expensive/slow regex/pattern parsing over a considerable amount of chunks (i.e. TB of logs or more) (e.g. parsing the access log from a high traffic gateway).
+
+Although the list is not exhaustive all examples granted can potentially result either in easily reaching limits (chunks, series, per-request, etc.) or take too long so that client applications and related networking infrastructure run in timeouts. Limits might be considered as tunable but timeouts might not be easy to mitigate or instantly addressable. These use cases can be considered problematic when the requesting user requires:
+
+* The query result without the ability to tune execution time or Loki's resource constraints.
+* The query result represents an expensive but important (aggregation) result set of logs and has a higher value as such irrespective how long it takes.
+
+## Goals
+
+The following proposals aim at providing the required infrastructure and query capabilities to execute queries in Loki asynchronously meaning the user will not fetch a result immediately but at later point in time using an identification for the requested query.
+
+## Non-Goals (optional)
+
+The following proposal do not aim at extending Loki to asynchronous capabilities beyond executing queries, e.g. offering asynchronous streams to read/write from/to Loki.
+
+## Proposals
+
+### Proposal 0: Do nothing
+
+The current state represents the do-nothing scenario very well, as per trade-offs currently to consider:
+
+* Users need to tune their queries manually (e.g. distinct label selectors, batching time frames).
+* Users need to request per-tenant and/or per-request limits by their respecting Loki administrators to overcome limits for certain query demand over time.
+* Client applications and networking infrastructure needs to be tuned for Loki response latency scenarios (e.g. Grafana datasource dataproxy, Server Write Timeouts on Gateways)
+
+### Proposal 1: Built-in asynchronous streaming component for asynchronous queries
+
+The main approach of a built-in asynchronous streaming component as devised in the [proof-of-concept PR](https://github.com/grafana/loki/pull/9730) is based on [nats.io's jetstream technology](https://nats.io/) and consists of:
+
+- A new Loki component called the `nats-server` that is hosting the persistence layer for any NATS JetStreams.
```

Comment:
I think we'll need some sort of abstraction. If we run the `all` target I expect all of this to happen in-memory or on temp file.